Document Campaign Builder canvas panning (#715)
mautic/user-documentation: docs/campaigns/campaign_builder.rst

Reviewer: ```suggestion
* **Contact Segments**: choose this option if you want to send your Campaign to a specific group of your Contacts that share certain attributes, for example, 'Located in the United States' or 'Visited Product A' and are in an existing Segment based on this criteria.

  Note that the Segment selection shows public Segments only. If you create a Segment marked as private, that Segment won't be available for use in Campaigns.

* **Contact Forms**: choose this option if you want to start the Campaign when the Contact completes a specified Form. Forms are the primary point of gathering information about a Contact. It's possible to take action in a Campaign based on the values submitted in the Form Field.

You can select a mix of both types of Contact sources for your Campaign. To use both, click the grey selector button on either the left or right side of the **Contact source** box to add whichever source type you didn't originally select.

.. note::

   To learn how to pan and move around the canvas, see :ref:`Navigating the canvas <navigating-the-canvas>`.
```

The Campaign Builder provides a blank canvas on which you can build your Campaign workflow. The Campaign Builder allows the use of conditions, decisions, and actions. It enables you to create a simple workflow by dragging and dropping various decisions, actions, and conditions onto a canvas.

.. _navigating-the-canvas:

Navigating the canvas

> ---------------------

Reviewer: After took a closer look, the header is jumping from H1 to H4. This isn't good for accessibility. But making this an H2 also seems make no sense.

My thoughts here:

- Add an H2 with title "Getting started with Campaign Builder". This will be a section that starts with 'To build your Campaign, perform the following steps:' in line 21.
- Move this section right after "Getting started with Campaign Builder" and make it an H3.
- Link this somewhere that make sense within "Getting started with Campaign Builder" to inform that there's a way to navigating the canvas.

The Campaign Builder provides a blank canvas on which you can build your Campaign workflow. The Campaign Builder allows the use of conditions, decisions, and actions. It enables you to create a simple workflow by dragging and dropping various decisions, actions, and conditions onto a canvas.

.. vale off

Getting started with Campaign Builder
*************************************

.. vale on

> To build your Campaign, perform the following steps. To learn how to pan and move around the canvas, see :ref:`Navigating the canvas <navigating-the-canvas>`.

Reviewer: I think we better link the "Navigating the canvas" in step 2 as a note admonition for better flow.

```suggestion
To build your Campaign, perform the following steps:
```

The Campaign Builder provides a blank canvas on which you can build your Campaign workflow. The Campaign Builder allows the use of conditions, decisions, and actions. It enables you to create a simple workflow by dragging and dropping various decisions, actions, and conditions onto a canvas.

.. _navigating-the-canvas:

Navigating the canvas
---------------------

> When you open the Campaign Builder, Mautic centers the view on your Campaign. The canvas uses an infinite-canvas interaction style without visible scrollbars, giving you more space for complex workflows.

Author: Added canvas panning documentation based on [PR #16127](https://github.com/mautic/mautic/pull/16127), which implements infinite-canvas style panning via `Mautic.initializeCampaignCanvasPanning()` in `campaign.js` and adds `grab`/`grabbing` cursor styles in `campaign.css`.

Source: https://github.com/mautic/mautic/pull/16127

Reviewer 2: Sounds great

> When you open the Campaign Builder, Mautic centers the view on your Campaign. The canvas uses an infinite-canvas interaction style without visible scrollbars, giving you more space for complex workflows.

Reviewer: ```suggestion
When opening the Campaign Builder, Mautic centers the view on the Campaign. The canvas uses an infinite-canvas interaction style, allowing free panning in any direction without traditional scrolling boundaries to give more space for complex workflows.
```

> To pan around the canvas, click and drag on any empty area of the canvas background. Your cursor changes to indicate when panning is active. You can navigate freely while still dragging nodes to position Campaign events.

Reviewer: ```suggestion
Click and drag on any empty area of the canvas background to pan around the workspace. The cursor changes to a grab icon during movement to indicate that panning is active. The canvas allows free navigation even while moving nodes to position Campaign events.
```

> To pan around the canvas, hold the spacebar and drag with your mouse. The cursor changes to a grab icon while panning. Release the spacebar to resume normal editing. This works like common design tools such as Figma or Photoshop, letting you navigate large Campaigns without accidentally selecting or moving individual nodes.

Reviewer: @promptless-for-oss add 'Photoshop', 'Figma', and 'spacebar' to `accept.txt` file.

   * **Contact Segments**: choose this option if you want to send your Campaign to a specific group of your Contacts that share certain attributes, for example, 'Located in the United States' or 'Visited Product A' and are in an existing Segment based on this criteria.

     Note that the Segment selection shows public Segments only. If you create a Segment marked as private, that Segment won't be available for use in Campaigns.

   * **Contact Forms**: choose this option if you want to start the Campaign when the Contact completes a specified Form. Forms are the primary point of gathering information about a Contact. It's possible to take action in a Campaign based on the values submitted in the Form Field.

   You can select a mix of both types of Contact sources for your Campaign. To use both, click the grey selector button on either the left or right side of the **Contact source** box to add whichever source type you didn't originally select.

>    .. note::

Reviewer: I think it would make more sense to make this a tip instead of a note?

>   For more information about Campaign Actions, Decisions, and Conditions, see the following topics.

Reviewer: As the next section is unrelated, we can remove this.

```suggestion
```

     - Checks if the Contact's Email address has a valid syntax, for example name@example.com without spaces, other invalid characters or formats.	
	
Notes on delayed conditions and dates
=====================================

Mautic respects delays set on the condition itself before passing down to a delay on any connected action. For example, if you are coming from a negative path on 'Opens Email', you can set a condition of 'has active notification' with a relative date of 1 day, followed by 'Send Email' on the negative path with a relative date of 2 days. Mautic checks after 1 day if there is an active notification and if there isn't, schedules the Email for two days later.

Using a custom date field to trigger a Campaign

> ===============================================

Reviewer: Wrap this in vale off/on statements.
